src/libaktualizr/package_manager/packagemanagerconfig.cc

```diff
@@ -54,11 +32,10 @@ void PackageConfig::writeToStream(std::ostream& out_stream) const {
   writeOption(out_stream, ostree_server, "ostree_server");
   writeOption(out_stream, packages_file, "packages_file");
   writeOption(out_stream, fake_need_reboot, "fake_need_reboot");
-#ifdef BUILD_DOCKERAPP
-  writeOption(out_stream, boost::algorithm::join(docker_apps, ","), "docker_apps");
-  writeOption(out_stream, docker_apps_root, "docker_apps_root");
-  writeOption(out_stream, docker_app_params, "docker_app_params");
-  writeOption(out_stream, docker_app_bin, "docker_app_bin");
-  writeOption(out_stream, docker_compose_bin, "docker_compose_bin");
-#endif
+
+  // note that this is imperfect as it will not print default values deduced
+  // from users of `extra`
+  for (const auto& e : extra) {
+    writeOption(out_stream, e.first, e.second);
+  }
```

Comment:
What's the issue? If you have extra package manager config options, and they are left unset, they get overlooked by the config dump? That is slightly annoying but it seems like we ought to be able to sort that out.

And I definitely approve of putting implementation-specific options in their own module instead of in the general config.

Reply:
I don't really know how to fix that here, since the `extra` won't even have been read at this point as `PackageConfig` knows nothing about its potential users.

Reply:
Yeah, sorry, didn't mean you have to fix it here and now! It might end up needing to be done elsewhere, but it seems minor enough to not worry about at present.